app-editors/atom/atom-1.11.2.ebuild

```diff
@@ -0,0 +1,398 @@
+
+	_s="${WORKDIR}/$(package_dir node-oniguruma)"
+	${EPYTHON} "${FILESDIR}/gyp-unbundle.py" \
+		--inplace --unbundle "onig_scanner;oniguruma;onig" \
+		"${_s}/binding.gyp" || die
+
+	_s="${WORKDIR}/$(package_dir node-spellchecker)"
+	${EPYTHON} "${FILESDIR}/gyp-unbundle.py" \
+		--inplace --unbundle "spellchecker;hunspell;hunspell" \
+		"${_s}/binding.gyp" || die
+
+	for binmod in ${BINMODS}; do
+		_s="${WORKDIR}/$(package_dir ${binmod})"
+		mkdir -p "${_s}/node_modules" || die
+		ln -s "${nan_s}" "${_s}/node_modules/nan" || die
+	done
+
+	# Unpack app.asar
+	easar extract "${S}/usr/share/atom/resources/app.asar" "${S}/build/app"
+
+	cd "${S}" || die
+
+	eapply "${FILESDIR}/atom-apm-path.patch"
+
+	sed -i -e "s|{{ATOM_SUFFIX}}|${suffix}|g" \
+		"${S}/build/app/src/config-schema.js" || die
+
+	eapply_user
+}
+
+src_configure() {
+	local binmod _s
+
+	for binmod in ${BINMODS}; do
+		einfo "Configuring ${binmod}..."
+		_s="${WORKDIR}/$(package_dir ${binmod})"
+		cd "${_s}" || die
+		enodegyp_atom configure
+	done
+}
+
+src_compile() {
+	local binmod _s x
+	local ctags_d="node_modules/symbols-view/vendor"
+	local jobs=$(makeopts_jobs) gypopts
+
+	gypopts="--verbose"
+
+	if [[ ${MAKEOPTS} == *-j* && ${jobs} != 999 ]]; then
+		gypopts+=" --jobs ${jobs}"
+	fi
+
+	mkdir -p "${S}/build/modules/" || die
+
+	for binmod in ${BINMODS}; do
+		einfo "Building ${binmod}..."
+		_s="${WORKDIR}/$(package_dir ${binmod})"
+		cd "${_s}" || die
+		enodegyp_atom ${gypopts} build
+		x=${binmod##node-}
+		mkdir -p "${S}/build/modules/${x}"
+		cp build/Release/*.node "${S}/build/modules/${x}"
+	done
+
+	# Put compiled binary modules in place
+	_fix_binmods "${S}/build" "app"
+	_fix_binmods "${S}/usr/share/atom/resources" "app"
+
+	# Remove non-Linux vendored ctags binaries
+	rm "${S}/build/app/${ctags_d}/ctags-darwin" \
+	   "${S}/build/app/${ctags_d}/ctags-win32.exe" || die
+
+	# Re-pack app.asar
+	# Keep unpack rules in sync with build/tasks/generate-asar-task.coffee
+	cd "${S}/build" || die
+	x="--unpack={*.node,ctags-config,ctags-linux,**/node_modules/spellchecker/**,**/resources/atom.png}"
+	easar pack "${x}" "app" "app.asar"
+	cd "${S}" || die
+}
+
+_fix_binmods() {
+	local _dir="${2}" _prefix="${1}" path relpath modpath mod depth link f d
+	local cruft
+
+	find "${_prefix}/${_dir}" -name '*.node' -print | while IFS= read -r path; do
+		f=$(basename "${path}")
+		d=$(dirname "${path}")
+	    relpath=${path#${_prefix}}
+		relpath=${relpath##/}
+		relpath=${relpath#W${_dir}}
+		modpath=$(dirname ${relpath})
+		modpath=${modpath%build/Release}
+		mod=$(basename ${modpath})
+
+		# must copy here as symlinks will cause the module loading to fail
+		cp -f "${S}/build/modules/${mod}/${f}" "${path}" || die
+		cruft=$(find "${d}" -name '*.a' -print)
+		if [ -n "${cruft}" ]; then
+			rm ${cruft} || die
+		fi
+	done
+}
+
+_fix_executables() {
+	local _dir="${1}" _node_sb="#!$(get_electron_dir)"/node
+
+	(find -L "${ED}/${_dir}" -maxdepth 1 -mindepth 1 -type f -print || die) \
```

Comment:
why do we require a subshell here? If it's required, why do we not use it in line 325?